src/Text/recipes.jl

```diff
@@ -0,0 +1,44 @@
+"""
+    TextFolders(textfile; labelfn = parentname, split = false)
+
+Recipe for loading a single-label text classification dataset
+stored in hierarchical folder format. 
+"""
+Base.@kwdef struct TextFolders <: Datasets.DatasetRecipe
+    labelfn = parentname
+    split::Bool = false
+    filefilterfn = _ -> true
+end
+
+Datasets.recipeblocks(::Type{TextFolders}) = Tuple{TextBlock, Label}
+
+function Datasets.loadrecipe(recipe::TextFolders, path)
+    isdir(path) || error("$path is not a directory")
+    data = loadfolderdata(
+        path,
+        filterfn=f -> istextfile(f) && recipe.filefilterfn(f),
+        loadfn=(loadfile, recipe.labelfn),
+        splitfn=recipe.split ? grandparentname : nothing)
+
+    (recipe.split ? length(data) > 0 : nobs(data) > 0) || error("No text files found in $path")
```

Comment:
Doesn't `nobs` fall back to `length` now (or vice versa)? Trying to understand why the nested ternary is necessary.

Reply:
Here, If the split is set to true then it checks for the corresponding loadfile mapobs and parentname mapobs by using `length(data)` which returns 2 when we have a valid `data` when using split. The next case would be checking for total obs in the data, done by `nobs(data)` return 50000 in this case. If none of this happens, we throw an error.

This is how I've seen it was implemented in `Vision/recipes.jl`.

Reply:
Got it. I think there needs to be a bit more documentation on this across recipes (or at least more well-named things in code so that one can figure out what all the `[2]` business is about, but that need not concern this PR.

Reply:
FastAI.jl does not use MLUtils.jl yet, see #196. Yes for the docs; in short, `blocks` is a tuple of `(inputblock, targetblock)` so `blocks[2]` takes the target block.